hls4ml/templates/vivado/firmware/defines.h

```diff
@@ -6,6 +6,9 @@
 #include "nnet_utils/nnet_types.h"
 #include <cstddef>
 #include <cstdio>
+#if __cplusplus > 202002L
```

Comment:
This check should be `#if __cplusplus >= 202302L` but since GCC doesn't fully support C++23 the constant is not set yet to that value, hence we check for "newer than C++20". Alternatively, we could check `__STDCPP_­BFLOAT16_­T__` and `__STDCPP_­FLOAT16_­T__` but these may also not be available even if the implementation is there...